frontend/src/pages/class/Insight.jsx

```diff
@@ -39,21 +38,21 @@ const columns = [
 ];
 
 const getRows = (studentResults) => {
-  const distribution = studentResults[0];
-  const studentInfo = studentResults[1];
-  const rows = [];
-  if (studentInfo && distribution) {
-    for (let i = 0; i < distribution.length; i++) {
+  if (studentResults) {
+    const distribution = studentResults[0];
+    const studentInfo = studentResults[1];
+    const rows = [];
+
+    distribution.forEach((d, index) => {
       rows.push({
-        id: studentInfo[i].id,
-        name: studentInfo[i].name,
-        correctness: distribution[i],
+        id: studentInfo[index].id,
+        name: studentInfo[index].name,
+        correctness: distribution[index],
```

Comment:
Als je `distribution.forEach((d, index) => ...)` doet en dan vervolgens `distribution[index]` aanroept is dat een beetje tegenstrijdig. Dus je kan hier gewoon `d` gebruiken. Misschien is het ook beter om dit een wat meer uitgeschreven naam te geven.

Reply:
ohja woops idk waarom ik dat deed